src/components/Tooltip/index.js

```diff
@@ -51,6 +51,9 @@ class Tooltip extends PureComponent {
      * @param {Object} bounds - updated bounds
      */
     updateBounds(bounds) {
+        if (bounds.width === 0) {
```

Comment:
> Can we add a comment explaining the purpose of this code?

I also thought of asking for a comment here but the line looks pretty self-explanatory no? If there is no width in the modal it means it shouldn't be rendered right?

Reply:
> The purpose of the function updateBounds was only to save the bounds

Ummm, that makes total sense : )

> This is clearly a side-effect. Now I think we should move this condition somewhere else.

@parasharrajat pretty sure you understand this much better than I do so can you help me understand **what problems does that line cause**?

Also, I'm not sure where else to add this condition if we really want to move this logic.

Reply:
@parasharrajat @PauloGasparSv as updateBounds is the place where we can apply **bounds.width** check because here we have direct access to this variable. So I think instead of dealing with multiple side effects like **componentDidUpdate**  and **updateBounds** we can go with the suggestion of @parasharrajat to rename this function, which will keep the rendering logic of the tooltip in one place.

Reply:
> Or we can rename this function if this place is convenient.

> we can go with the suggestion of @parasharrajat to rename this function, which will keep the rendering logic of the tooltip in one place.

Great, let's please change that function name then :D

Tks again for the help and explanation @parasharrajat

Reply:
Since everything related to this issue is closed and paid out and the whole review process is a bit lengthy I think we should rename that in a new P.R. but not necessarily create a new one.

I'll try to fix this next time I work on an App P.R. since it's not a priority.